.github/calcrom/calcrom.cpp

```diff
@@ -19,135 +23,82 @@
-            throw runtime_error(ss.str());
-        }
-        assign(glob_result.gl_pathv, glob_result.gl_pathv + glob_result.gl_pathc);
-    };
-    void operator~() {
-        globfree(&glob_result);
-    }
+    missing_option(string& error) : invalid_argument{error.c_str()} {}
 };
 
-void analyze(string basedir, string subdir, string version) {
-    fstream elf;
-    Elf32_Ehdr ehdr;
-    vector<Elf32_Shdr> shdr;
-    stringstream pattern;
-
-    // Accumulate sizes
-    //        src   asm
-    // data  _____|_____
-    // text       |
-    unsigned sizes[2][2] = {{0, 0}, {0, 0}};
-    char * shstrtab = NULL;
-    size_t shstrsz = 0;
-    stringstream builddir;
-    builddir << subdir << "/build/" << version;
-    pattern << basedir << "/" << subdir << "/{src,asm,lib/src,lib/{libc,libnns,NitroSDK}/src,modules/*/{src,asm}}/*.{c,s,cpp}";
-    for (char const * & fname : Glob(pattern.str()))
-    {
-        string fname_s(fname);
-        string ext = fname_s.substr(fname_s.rfind('.'), 4);
-        bool is_asm = ext == ".s";
-        fname_s = fname_s.replace(fname_s.find(subdir), 4, builddir.str());
-        fname_s = fname_s.replace(fname_s.rfind('.'), 4, ".o");
-        elf.open(fname_s, ios_base::in | ios_base::binary);
-        if (!elf.good()) {
-            cerr << "Error: file not found: " << fname_s << endl;
-            return;
-        }
-        elf.read((char *)&ehdr, sizeof(ehdr));
-        if (memcmp(ehdr.e_ident, ELFMAG, SELFMAG) != 0
-            || ehdr.e_ehsize != sizeof(Elf32_Ehdr)
-            || ehdr.e_shentsize != sizeof(Elf32_Shdr))
-        {
-            elf.close();
-            stringstream ss;
-            ss << "Error validating " << fname_s << " as an ELF file" << endl;
-            throw runtime_error(ss.str());
-        }
-        // Read ELF sections
-        elf.seekg(ehdr.e_shoff);
-        shdr.resize(ehdr.e_shnum);
-        elf.read((char *)shdr.data(), ehdr.e_shnum * ehdr.e_shentsize);
-
-        // Read .shstrtab
-        if (shstrsz < shdr[ehdr.e_shstrndx].sh_size) {
-            shstrtab = (char *)realloc(shstrtab, shdr[ehdr.e_shstrndx].sh_size);
-            shstrsz = shdr[ehdr.e_shstrndx].sh_size;
-        }
-        elf.seekg(shdr[ehdr.e_shstrndx].sh_offset);
-        elf.read(shstrtab, shdr[ehdr.e_shstrndx].sh_size);
-        elf.close();
-
-        // Analyze sections
-        for (Elf32_Shdr & hdr : shdr) {
-            string shname = shstrtab + hdr.sh_name;
-            bool is_text = (shname == ".text" || shname == ".init" || shname == ".itcm");
-            bool is_data = (shname == ".data" || shname == ".rodata" || shname == ".sdata" || shname == ".dtcm");
-            size_t size = hdr.sh_size + (hdr.sh_size & 3 ? 4 - (hdr.sh_size & 3) : 0);
-            if (is_text || is_data)
-            {
-                sizes[is_text][is_asm] += size;
+struct Options {
+    path arm9subdir = "";
+    path arm7subdir = "sub";
```

Comment:
```suggestion
    path arm9subdir = "arm9";
    path arm7subdir = "arm7";
```

Reply:
The corresponding change was made in .github/calcrom/webhook.sh#L19